uint8 level
string text
